crates/compiler/specialize_types/src/mono_num.rs

```diff
@@ -0,0 +1,16 @@
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Number {
+    I8(i8),
+    U8(u8),
+    I16(i16),
+    U16(u16),
+    I32(i32),
+    U32(u32),
+    I64(i64),
+    U64(u64),
+    I128(i128),
+    U128(u128),
+    F32(f32),
+    F64(f64),
+    Dec(f64),
```

Comment:
@rtfeldman is this intentional?

Reply:
For now, yeah. It's how we have it in earlier stages of compilation - I think because we didn't have a `&str -> Dec` parser at the time, but we did have `f64 -> Dec`.

When we change that at the earlier stage we can update it here too!